transitive_maven_jar/transitive_maven_jar.bzl

```diff
@@ -0,0 +1,55 @@
+
+DEPS = ["mvn"]
+MAX_TIMEOUT = 0x7FFFFFFF
+
+BINARY_URL = "https://github.com/petroseskinder/migration-tooling/releases/download/test/generate_workspace_deploy.jar"
+
+def _check_dependencies(rctx):
+	for dep in DEPS:
+		if rctx.which(dep) == None:
+			fail("%s requires %s as a dependency. Please check your PATH." % (rctx.name, dep))
+
+def _validate_coordinate_length(coordinate):
+	parts = coordinate.split(":")
+	return len(parts) >= 2 and len(parts) <= 5
+
+def _validate_coordinates(rctx):
+	coordinates = rctx.attr.artifacts
+	for coord in coordinates:
+		if _validate_coordinate_length(coord) == False:
+			fail("Invalid coordinate %s. Generally formatted as \"group:artifact:version\"" % coord)
+	return True
+
+def _create_arguments(rctx):
+	#TODO(petros): pass output directory as an attribute.
+	arguments = ['-a ' + artifact for artifact in rctx.attr.artifacts]
```

Comment:
Might as well go for readability here and do `--artifact`.

You should be able to use [`rctx.path()`](https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#path) to get the output dir.